be/src/vec/core/sort_block.h

```diff
@@ -259,15 +261,25 @@ class ColumnSorter {
                         flags[not_null_range.first] = 0;
                     }
                     if (is_null_range.first < is_null_range.second) {
+                        // do not sort null values
                         std::fill(flags.begin() + is_null_range.first,
-                                  flags.begin() + is_null_range.second, 1);
+                                  flags.begin() + is_null_range.second, 0);
 
-                        flags[is_null_range.first] = 0;
+                        if (_limit > is_null_range.first && _limit <= is_null_range.second) {
+                            _limit = is_null_range.second;
```

Comment:
`_limit > is_null_range.first && _limit <= is_null_range.second` will be true only once so I think this line is trival and can be removed

Reply:
Why can this statement be removed? I think an `UNLIKELY` could be added.